File: Assets/Scripts/ScoreList/ScoreView.js
class ScoreView implements View {

    public final static var NAME : String = "Score";

    public var guiSkin : GUISkin;
    
    public var header1Style : GUIStyle;
    public var header2Style : GUIStyle;
    public var header2ErrorStyle : GUIStyle;
    public var formFieldStyle : GUIStyle;
    
    public var error = false;
    public var errorMessage : String = "";
    
    public var enterGameHandler : function();
    public var exitGameHandler : function();

    private var blockUI 		= false;
    private var isMyScoreOnly 	= false;
    private var isRankMode		= false;
    private var scores : System.Collections.Generic.List.<GBaaS.io.Objects.GBScoreObject>;
 
 	function getScore() {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	script.Init();
    	
    	if(isMyScoreOnly) {
    		scores = script.GetScore("1", "point", 10, true, false);
    	} else {
    		scores = script.GetScore("1", "point", 10, false, false);
    	}
    	
       	Debug.Log("In getScore of ScoreView result : " + scores.ToString());
 	}
 	
 	 function getRank() {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	script.Init();
    	
    	scores = script.GetRank("1", "point", "DESC", "WEEKLY", 0, 10);
    	
       	Debug.Log("In getRank of ScoreView result : " + scores.ToString());
 	}
 	
 	function getScoreBuf() {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	
    	//Debug.Log("in getScoreBuf script is : " + script.ToString());
    	
    	script.Init();
    	scores = script.GetScoreBuf();
 	}
 	
 	function scoreBufReset() {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	
    	script.ScoreBufReset();
 	}
 	
 	function pushNotify(msg) {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	
    	script.PushNotify(msg);
 	}

 	function logout() {
 		var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	
    	script.Init();
    	script.Logout();
 	}
 	
    function render() {

        var screenWidth = Screen.width;
        var screenHeight = Screen.height;
    
        var xShift = (screenWidth - 260)/2;
        var yShift = (screenHeight - 600)/2;
       
        GUI.skin = guiSkin;
        
        // Disabling UI if blockUI is true: 
        GUI.enabled = !blockUI;

        // Main label:
        GUI.Label(Rect(0, yShift, screenWidth, 30), "지바맨 랭킹 TOP 10", header1Style);
        
        getScoreBuf();
        if(scores == null) return;
        
        // Message label:
        if(error) {
            GUI.Label(Rect(0, yShift + 70, screenWidth, 30), errorMessage, header2ErrorStyle);
        } else {      
        	if(this.isRankMode) {
    		  	for(var i=0; i<scores.Count; i++) {
		       		GUI.Label(Rect(0, yShift + 70 + (30 * i), screenWidth, 30), "[" + scores[i].rank + "] " + scores[i].displayName + " : " + scores[i].score, header2Style);
		        }
        	} else {
		       	for(var j=0; j<scores.Count; j++) {
		       		GUI.Label(Rect(0, yShift + 70 + (30 * j), screenWidth, 30), scores[j].displayName + " : " + scores[j].score, header2Style);
		        }
			}
        }
        
        // Login button:
        if(GUI.Button(Rect(xShift, yShift + 420, 120, 30), "게임하기")) {
        	scoreBufReset();
            enterGameHandler();
        }
       
        // Switch to registration view button:
        if(GUI.Button(Rect(xShift + 140, yShift + 420, 120, 30), "나가기")) {
			Debug.Log("In Button Exit");
        	scoreBufReset();
        	logout();
            exitGameHandler();
        }
        
        // Login button:
        if(GUI.Button(Rect(xShift, yShift + 460, 260, 30), "자랑하기(push)")) {
            pushNotify("지바맨2 점수자랑");
        }
        
        // Login button:
        if(GUI.Button(Rect(xShift, yShift + 500, 120, 30), "내점수보기")) {
            setMyScoreMode(true);
            setRankMode(false);
            getScore();
        }
       
        // Switch to registration view button:
        if(GUI.Button(Rect(xShift + 140, yShift + 500, 120, 30), "최고점수보기")) {
            setMyScoreMode(false);
            setRankMode(false);
            getScore();
        }

        if(GUI.Button(Rect(xShift + 140, yShift + 540, 120, 30), "내등수보기")) {
            setMyScoreMode(false);
            setRankMode(true);
            getRank();
        }

        // Enabling UI: 

        GUI.enabled = true;
    }

    public function setBlockUI(blockUI) {
        this.blockUI = blockUI;
    }
    
    public function setMyScoreMode(boolValue) {
        this.isMyScoreOnly = boolValue;
    }
    
    public function setRankMode(boolValue) {
    	this.isRankMode = boolValue;
    }
}

File: Assets/Scripts/ScoreList/ScoreView.cs
using UnityEngine;
using System.Collections.Generic;

using GBaaS.io;
using GBaaS.io.Objects;

public interface ScoreViewListener {
	void OnEnterGame();
	void OnExitGame();
}

class ScoreView : GBaaSApiHandler, View {

    public const string NAME = "Score";

	public GUISkin guiSkin;
    
	public GUIStyle header1Style;
	public GUIStyle header2Style;
	public GUIStyle header2ErrorStyle;
	public GUIStyle formFieldStyle;
    
    public bool 	error 			= false;
    public string 	errorMessage 	= "";
    
	private bool 	blockUI 		= false;

	private List<GBScoreObject> _scores = null;
 
	private ScoreViewListener _listener = null;
	
	public void SetListener(ScoreViewListener listener) {
		_listener = listener;
	}
	
	public void enterGameHandler() {
		if(_listener != null) {
			_listener.OnEnterGame();
		}
	}

	public void exitGameHandler() {
		if(_listener != null) {
			_listener.OnExitGame();
		}
	}
 	
	public override void OnGetScore(List<GBScoreObject> result) {
		if(result[0].isSuccess) {
			Debug.Log("ScoreView OnGetScore result : " + _scores.Count.ToString());
			_scores = result;
		} else {
			Debug.Log("ScoreView OnGetScore Fail returnCode(" + result[0].returnCode + ") reason(" + result[0].reason + ")");
		}
	}

	public override void OnGetScoreLog(List<GBScoreObject> result) {
		_scores = result;
		Debug.Log("ScoreView OnGetScoreLog result : " + _scores.Count.ToString());
	}
	
	public override void OnGetRank(List<GBScoreObject> result) {
		_scores = result;
		Debug.Log("ScoreView OnGetRank result : " + _scores.Count.ToString());
	}
	 	
 	public void pushNotify(string msg) {
		GBaaSObject.Instance.Init(this);
		GBaaSObject.Instance.PushNotify(msg);
 	}

 	public void logout() {
		GBaaSObject.Instance.Init(this);
		GBaaSObject.Instance.API.Logout();
 	}
 	
    public void render() {

        int screenWidth = Screen.width;
        int screenHeight = Screen.height;
    
        float xShift = (screenWidth - 260)/2;
        float yShift = (screenHeight - 600)/2;
       
        GUI.skin = guiSkin;
        
        // Disabling UI if blockUI is true: 
        GUI.enabled = !blockUI;

        // Main label:
        GUI.Label(new Rect(0, yShift, screenWidth, 30), "지바맨 랭킹 TOP 10", header1Style);
        
        if(_scores == null) {
			Debug.Log("In render Score == null");
			_scores = new List<GBScoreObject>();
			GBaaSObject.Instance.Init(this);
			GBaaSObject.Instance.API.GetScore("1", "point", 10);
			return;
		}

        // Message label:
        if(error) {
            GUI.Label(new Rect(0, yShift + 70, screenWidth, 30), errorMessage, header2ErrorStyle);
        } else {     
			bool isRankMode = false;
			if(_scores.Count > 0) {
				isRankMode = (_scores[0].rank > 0);
			}

			float yPos = yShift + 40;
			GUI.Label(new Rect(xShift - 320, yPos, 150, 30), "rank", header2Style);
			GUI.Label(new Rect(xShift - 170, yPos, 150, 30), "diaplayName", header2Style);
			GUI.Label(new Rect(xShift - 20, yPos, 150, 30), "username(ID)", header2Style);
			GUI.Label(new Rect(xShift + 130, yPos, 150, 30), "stage", header2Style);
			GUI.Label(new Rect(xShift + 280, yPos, 150, 30), "score", header2Style);
			GUI.Label(new Rect(xShift + 430, yPos, 150, 30), "unit", header2Style);

			for(var i=0; i<_scores.Count; i++) {
				yPos = yShift + 70 + (30 * i);
				if(isRankMode) GUI.Label(new Rect(xShift - 320, yPos, 150, 30), _scores[i].rank.ToString(), header2Style);
				GUI.Label(new Rect(xShift - 170, yPos, 150, 30), _scores[i].displayName, header2Style);
				if(_scores[i].username.Length > 10) {
					GUI.Label(new Rect(xShift - 20, yPos, 150, 30), _scores[i].username.Substring(0, 10) + "...", header2Style);
				} else {
					GUI.Label(new Rect(xShift - 20, yPos, 150, 30), _scores[i].username.Substring(0, _scores[i].username.Length), header2Style);
				}
				GUI.Label(new Rect(xShift + 130, yPos, 150, 30), _scores[i].stage, header2Style);
				GUI.Label(new Rect(xShift + 280, yPos, 150, 30), _scores[i].score.ToString(), header2Style);
				GUI.Label(new Rect(xShift + 430, yPos, 150, 30), _scores[i].unit, header2Style);
	        }
        }
        
        if(GUI.Button(new Rect(xShift, yShift + 420, 120, 30), "게임하기")) {
            enterGameHandler();
        }
       
        if(GUI.Button(new Rect(xShift + 140, yShift + 420, 120, 30), "나가기")) {
			Debug.Log("In Button Exit");
        	logout();
            exitGameHandler();
        }
        
        if(GUI.Button(new Rect(xShift, yShift + 460, 260, 30), "자랑하기(push)")) {
            pushNotify("지바맨2 점수자랑");
        }
        
		if(GUI.Button(new Rect(xShift, yShift + 500, 120, 30), "자기점수기록")) {
			GBaaSObject.Instance.API.GetScoreLog("1", "point", 10);
		}
		
		if(GUI.Button(new Rect(xShift + 140, yShift + 500, 120, 30), "상위등수보기")) {
			GBaaSObject.Instance.API.GetScore("1", "point", 10);
        }

		if(GUI.Button(new Rect(xShift, yShift + 540, 120, 30), "자기등수")) {
			GBaaSObject.Instance.API.GetRank("", "", ScoreOrder.DESC, Period.Monthly);
		}

		if(GUI.Button(new Rect(xShift + 140, yShift + 540, 120, 30), "자기등수주변")) {
			GBaaSObject.Instance.API.GetRank("", "", ScoreOrder.DESC, Period.Monthly, 0, 10);
		}
		
        // Enabling UI: 

        GUI.enabled = true;
    }

    public void setBlockUI(bool blockUI) {
        this.blockUI = blockUI;
    }
}
